lib/plugins/aws/deploy/lib/uploadArtifacts.js

```diff
@@ -9,19 +9,40 @@ const filesize = require('filesize');
 const normalizeFiles = require('../../lib/normalizeFiles');
 const getLambdaLayerArtifactPath = require('../../utils/getLambdaLayerArtifactPath');
 const ServerlessError = require('../../../../serverless-error');
+const { legacy, progress } = require('@serverless/utils/log');
 
 const MAX_CONCURRENT_ARTIFACTS_UPLOADS =
   Number(process.env.SLS_MAX_CONCURRENT_ARTIFACTS_UPLOADS) || 3;
 
 module.exports = {
   async uploadArtifacts() {
+    const artifactFilePaths = [
+      ...this.getFunctionArtifactFilePaths(),
+      ...this.getLayerArtifactFilePaths(),
+    ];
+    if (artifactFilePaths.length === 1) {
+      const stats = (() => {
+        try {
+          return fs.statSync(artifactFilePaths[0]);
+        } catch (error) {
+          throw new ServerlessError(
+            `Cannot read file artifact "${artifactFilePaths[0]}": ${error.message}`,
+            'INACCESSIBLE_FILE_ARTIFACT'
+          );
+        }
+      })();
```

Comment:
Why this logic is duplicated here (it's also in context `uploadFunctionsAndLayers`) ?

Reply:
good spot, I'll refactor it to a separate util :+1:

Reply:
Still problem here is, not only that logic is in two places, but that it's triggered twice I think.

I think just removing it from here will fix both issues

Reply:
We cannot really remove it from here because otherwise, we don't know the size that we should show in the main spinner if we upload only single file. I guess we could maybe have it as util with memoization to avoid unnecessary fs operations?

Reply:
changed - for now I'm keeping the util in `uploadArtifacts`, but it's likely I will move it when implementing the package size display for functions in summary